test/libsolidity/syntaxTests/errors/error_param_type_shadowed_by_param_name.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    enum EnumType {A, B, C}
+
+    struct StructType {
+        uint x;
+    }
+
+    error E1(StructType StructType);
+    error E2(EnumType EnumType);
+    error E3(EnumType StrucType, StructType EnumType);
+}
+// ----
```

Comment:
Does this test pass as it is?

Reply:
It does. There's no ambiguity because you cannot define new types inside an error declaration. Allowing parameter names to shadow types would never lead to valid code so the PR makes name checker not consider them.

BTW, I just noticed a typo. `EnumType StrucType` should be `EnumType StructType`. But it should still pass with that if the PR is properly implemented.

Reply:
Fixed.